data/provider/src/main/java/tech/pegasys/teku/api/blobselector/BlobSidecarSelectorFactory.java

```diff
@@ -84,6 +94,16 @@ private SafeFuture<Optional<List<BlobSidecar>>> getBlobSidecarsForBlock(
       return SafeFuture.completedFuture(Optional.empty());
     }
     final SignedBeaconBlock block = maybeBlock.get();
-    return client.getBlobSidecars(block.getSlotAndBlockRoot(), indices).thenApply(Optional::of);
+    return getBlobSidecars(block.getSlotAndBlockRoot(), indices);
+  }
+
+  private SafeFuture<Optional<List<BlobSidecar>>> getBlobSidecars(
+      final SlotAndBlockRoot slotAndBlockRoot, final List<UInt64> indices) {
+    return client.getBlobSidecars(slotAndBlockRoot, indices).thenApply(Optional::of);
+  }
+
+  private SafeFuture<Optional<List<BlobSidecar>>> getBlobSidecars(
```

Comment:
maybe `getFinalizedBlobSidecars` for more clarity?

Reply:
I added comment instead